dipy/io/statefull_tractogram.py

```diff
@@ -0,0 +1,879 @@
+                    trk_header_check=True):
+    """ Applies median filter multiple times on input data.
+
+    Parameters
+    ----------
+    filename : string
+        Filename with valid extension
+    reference : Nifti or Trk filename, Nifti1Image or TrkFile, Nifti1Header or
+        trk.header (dict)
+        Reference that provides the spatial attribute.
+        Typically a nifti-related object from the native diffusion used for
+        streamlines generation
+    space : string
+        Space in which the streamlines will be transformed after loading
+        (vox, voxmm or rasmm)
+    shifted_origin : bool
+        Information on the position of the origin,
+        False is Trackvis standard, default (center of the voxel)
+        True is NIFTI standard (corner of the voxel)
+
+    Returns
+    -------
+    output : StateFullTractogram
+        The tractogram to load (must have been saved properly)
+    """
+    _, extension = robust_split_name(filename)
+    if extension not in ['.trk', '.tck', '.vtk', '.fib', '.dpy']:
+        logging.error('Output filename is not one of the supported format')
+        return False
+
+    if to_space not in Space:
+        logging.error('Space MUST be one of the 3 choices (Enum)')
+        return False
+
+    if trk_header_check and extension == '.trk':
+        if not is_header_compatible(filename, reference):
+            logging.error('Trk file header does not match the provided ' +
+                          'reference')
+            return False
+
+    timer = time.time()
+    data_per_point = None
+    data_per_streamline = None
+    if extension in ['.trk', '.tck']:
+        tractogram_obj = nib.streamlines.load(filename).tractogram
+        streamlines = tractogram_obj.streamlines
+        if extension == '.trk':
+            data_per_point = tractogram_obj.data_per_point
+            data_per_streamline = tractogram_obj.data_per_streamline
+
+    elif extension in ['.vtk', '.fib']:
+        streamlines = load_vtk_streamlines(filename)
+    elif extension in ['.dpy']:
+        dpy_obj = Dpy(filename, mode='r')
+        streamlines = list(dpy_obj.read_tracks())
+    logging.debug('Load %s with %s streamlines in %s seconds',
+                  filename, len(streamlines), round(time.time() - timer, 3))
+
+    sft = StateFullTractogram(streamlines, reference, Space.RASMM,
+                              shifted_origin=shifted_origin,
+                              data_per_point=data_per_point,
+                              data_per_streamline=data_per_streamline)
+
+    if to_space == Space.VOX:
+        sft.to_vox()
+    elif to_space == Space.VOXMM:
+        sft.to_voxmm()
+
+    if bbox_valid_check and not sft.is_bbox_in_vox_valid():
+        raise ValueError('Bounding box is not valid in voxel space, cannot ' +
+                         'load a valid file if some coordinates are invalid')
+
+    return sft
+
+
+def robust_split_name(filename):
```

Comment:
This function is more broadly useful and should be in the dipy.io base namespace!

Reply:
I'd delegate this to Nibabel. See my other comment about dealing with filename extensions.